src/StreamJsonRpc/JsonRpc.cs

```diff
@@ -942,6 +947,26 @@ protected virtual void Dispose(bool disposing)
         /// </remarks>
         protected virtual bool IsFatalException(Exception ex) => false;
 
+        /// <summary>
+        /// Creates the <see cref="JsonRpcError.ErrorDetail"/> to be used as the value for the error property to be sent back to the client in response to an exception being thrown from an RPC method invoked locally.
+        /// </summary>
+        /// <param name="request">The request that led to the invocation that ended up failing.</param>
+        /// <param name="exception">The exception thrown from the RPC method.</param>
+        /// <returns>The error details to return to the client. Must not be <c>null</c>.</returns>
+        /// <remarks>
+        /// This method may be overridden in a derived class to change the way error details are expressed.
+        /// </remarks>
+        protected virtual JsonRpcError.ErrorDetail CreateErrorDetails(JsonRpcRequest request, Exception exception)
+        {
+            var localRpcEx = exception as LocalRpcException;
+            return new JsonRpcError.ErrorDetail
+            {
+                Code = (JsonRpcErrorCode?)localRpcEx?.ErrorCode ?? JsonRpcErrorCode.InvocationError,
+                Message = exception.Message,
+                Data = localRpcEx != null ? localRpcEx.ErrorData : new CommonErrorData(exception),
```

Comment:
Can ErrorData be null? If so, does it matter if that's what returned for Data?

Reply:
Yes, the error.data property [may be null](https://www.jsonrpc.org/specification#error_object). And yes, the intent here is that if the special exception is thrown that gives control over the data object, that we honor it, even if it is null.